test/e2e/prometheus_test.go

```diff
@@ -3702,13 +3652,6 @@ func testPromSecurePodMonitor(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			prom := framework.MakeBasicPrometheus(ns, name, name, 1)
-			prom.Namespace = ns
-
-			if _, err := framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, prom); err != nil {
-				t.Fatal(err)
-			}
-
```

Comment:
Creating the service monitor before Prometheus speeds up the test because Prometheus will start scraping the target immediately instead of waiting for the configuration update to propagate to the Prometheus secret (and trigger a config reload).